src/components/Modals/NewMessageModal.jsx

```diff
@@ -0,0 +1,215 @@
+      setError('Please enter a value for Message Title');
+    } else if (!message.recipientPodUrl) {
+      setError('Please enter a value for Recipient Pod URL');
+    } else if (!message.message) {
+      setError('Please enter a value for the Message');
+    } else {
+      try {
+        await sendMessageTTL(session, message, podUrl);
+
+        setMessage({
+          recipientPodUrl: '',
+          title: '',
+          message: ''
+        });
+        setError('');
+        setSuccess(`Message successfully sent to ${message.recipientPodUrl}`);
+        setSuccessTimeout(true);
+        setTimeout(() => {
+          setSuccessTimeout(false);
+        }, 10000);
+      } catch (err) {
+        // TODO: Make sure invalid username is the only possible error
+        setError(err.message);
+      } finally {
+        setOriginalMessage('');
+      }
+    }
+
+    // Re-sorts messages when new message is added to outboxList
+    const outboxMessages = await getMessageTTL(session, 'Outbox', outboxList, podUrl);
+    const sortedOutbox = outboxMessages;
+    sortedOutbox.sort((a, b) => b.uploadDate - a.uploadDate);
+    setOutboxList(sortedOutbox);
+  };
+
+  /* eslint-disable jsx-a11y/label-has-associated-control */
+  return (
+    <Dialog
+      open={showModal}
+      aria-labelledby="new-message-modal"
+      onClose={() => setShowModal(false)}
+    >
+      <Box
+        noValidate
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'center',
+          alignItems: 'center',
+          padding: '20px',
+          minWidth: '50%',
+          minHeight: '90%'
+        }}
+      >
+        <form onSubmit={(e) => handleSubmit(e)} autoComplete="off">
+          <Typography display="flex" justifyContent="center" variant="h5">
+            {oldMessage ? 'Reply To' : 'New Message'}
+          </Typography>
+          <TextField
+            margin="normal"
+            value={message.recipientPodUrl}
+            type="text"
+            name="recipientPodUrl"
+            id="recipientPodUrl"
+            onChange={(e) => handleChange(e)}
+            required
+            autoFocus
+            label="To"
+            fullWidth
+          />
+          <TextField
+            margin="normal"
+            value={message.title}
+            type="text"
+            name="title"
+            id="title"
+            onChange={(e) => handleChange(e)}
+            required
+            label="Subject"
+            inputProps={{ maxLength: '24' }}
+            fullWidth
+          />
+          <TextField
+            margin="normal"
+            value={oldMessage && originalMessage}
+            type="previousMessage"
+            name="previousMessage"
+            id="previousMessage"
+            label="Previous Message"
+            // helperText="Previous message"
+            // variant="filled"
+            multiline
+            rows={3}
+            InputProps={{
```

Comment:
Would have to test (when you have an old message and reply to it), but I would think using the prop `disabled` would be better than this `read-only`.  
Personally, don't like that you can still click on/focus on this input with read-only, but I don't know what is the acceptable standard for accessibility in this case.